### DISCUSSION TITLE
Edit dashboard JSON not working when UID removed (creates an entirely new dashboard 🤯)

### DISCUSSION THREAD

**Discussion by samjewell** [grafana/grafana#40099] [Dashboards, Panels & Visualizations]
I generated some Dashboard JSON using Jsonnet, and used /dashboard/import to create a new dashboard from it. Then I realise a problem in my JSON (the datasource needs to be changed), so I update my Jsonnet, regenerate my JSON, and then try to update the JSON for the dashboard that I've just created.

On the dashboard I just created I do the following:
- Visit "Dashboard Settings" then "JSON Model"
- Paste in the JSON that I just rendered from my fixed Jsonnet (**which has no uid present**)
- Click Save changes

A tiny JSON file which reproduces the issue is as follows:
<details>
<summary><strong>JSON file for reproduction</strong></summary>
<br />

```json
{
  "__inputs": [
    {
      "description": "",
      "label": "prometheus",
      "name": "DS_PROMETHEUS",
      "pluginId": "prometheus",
      "pluginName": "Prometheus",
      "type": "datasource"
    }
  ],
  "panels": [
    {
      "datasource": "grafanacloud-usage",
      "fieldConfig": {
        "defaults": {
          "color": {
            "mode": "thresholds"
          },
          "mappings": [],
          "thresholds": {
            "mode": "absolute",
            "steps": [
              {
                "color": "green",
                "value": null
              }
            ]
          }
        },
        "overrides": []
      },
      "gridPos": {
        "h": 9,
        "w": 12,
        "x": 0,
        "y": 0
      },
      "id": 2,
      "options": {
        "colorMode": "value",
        "graphMode": "area",
        "justifyMode": "auto",
        "orientation": "auto",
        "reduceOptions": {
          "calcs": [
            "lastNotNull"
          ],
          "fields": "",
          "values": false
        },
        "text": {},
        "textMode": "auto"
      },
      "pluginVersion": "8.3.0-36021pre",
      "targets": [
        {
          "exemplar": true,
          "expr": "1",
          "instant": false,
          "interval": "",
          "legendFormat": "partners",
          "refId": "A"
        },
        {
          "exemplar": true,
          "expr": "2",
          "format": "time_series",
          "hide": false,
          "instant": false,
          "interval": "",
          "legendFormat": "parents",
          "refId": "B"
        }
      ],
      "title": "Family",
      "type": "stat"
    }
  ],
  "title": "Samily-Family",
  "version": 1
}
```

</details>

---

**Observed behaviour:**
1. (Without any changes to the dashboard title) Save fails, and an error is shown "A dashboard with the same name in the folder already exists"
2. If I make changes to the dashboard title: The page does a quick reload and no errors are shown, so I assume that the save worked. However I'm still looking at the old dashboard, showing the old dashboard JSON! The JSON I just pasted and saved is no longer in the JSON editable panel in front of me - I'm now looking at the old version, wondering where the hell my paste just went. Later if I'm lucky enough to explore in the right places, I can discover that the JSON I pasted has actually landed in an entirely new dashboard in my General folder - in my case named "Samily-Family2". But it took me a full working day to discover this, and the help of my pair-programmer, in reality!

**Expected behaviour: One of the following:**
1. Update the existing dashboard with the new JSON, retaining the same dashboard uid
2. At least show a warning; warn me that a new dashboard is being created instead of the current one being updated, due to the lack of UID in the pasted JSON, and the fact that the default behaviour is to create a new UID when none is present
3. Actually route me properly through the `/dashboard/import` flow, and show me the "Options" page between taking my pasted JSON and creating the dashboard. That way I can resolve any naming collisions as I see fit, or choose to overwrite if that's what I want to do instead. Added bonus of this approach - if the JSON includes a one-off dashboard creation configuration flag such as `__inputs: […]`, then this flag will be respected and (for example, as per the JSON example given above) the user asked to choose which Datasource they want to use for that dashboard.
   <img src=https://user-images.githubusercontent.com/2903904/136241042-a130e756-1837-4adc-8c30-92a7827125e6.png width=200>

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!

**Comment by samjewell** [comment]
I think it's quite hard to see where this goes in the Github UI:
![image](https://github.com/grafana/grafana/assets/2903904/35aa7725-eb1c-470e-ba2b-b72d42e47555)

I opened an issue for this here #82833
